lib/src/rules/maxLength.ts

```diff
@@ -0,0 +1,24 @@
+import { FieldValidationResult, FieldValidationFunction } from '../entities';
+import {
+  LengthParams,
+  parseLengthParams,
+  isLengthValid,
+} from './length';
+
+const DEFAULT_PARAMS = { length: undefined };
+const BAD_PARAMETER = 'FieldValidationError: Parameter "length" for maxLength in customParams is mandatory and should be a valid number. Example: { length: 4 }.';
+
+export function maxLength(value: string, vm, customParams: LengthParams = DEFAULT_PARAMS) {
+  const length = parseLengthParams(customParams, BAD_PARAMETER);
+  const isValid = isLengthValid(value, length, isStringLengthValid);
+  const validationResult = new FieldValidationResult();
+
+  validationResult.succeeded = isValid;
+  validationResult.key = 'MAX_LENGTH';
+  validationResult.errorMessage = isValid ? '' : `The value provided is too long. Length must not exceed ${length} characters.`;
+  return validationResult;
+}
+
+function isStringLengthValid(value: string, length: number) {
+  return isNaN(length) ? false : value.length <= length;
```

Comment:
- It's necessary check that length isNaN?
Yes. Length must be a number.
- What about check value is != null or undefined?
`isNaN` does implicit `Number(<value>)`.
- To improve readability is better introduce a carriage return after `?` and `:`.
Do you mean insert (CR)/LF after `return` statement like this?:

```ts
function isStringLengthValid(value: string, length: number) {
  return isNaN(length) ? false : value.length <= length;

}
```

That has no sense to me.